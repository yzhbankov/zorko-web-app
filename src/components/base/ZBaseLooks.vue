<template>
  <v-container fluid v-bind="{ [`grid-list-${size}`]: true }">
    <v-layout row wrap>
      <v-flex
        v-for="item in items"
        :key="item.name"
        xs4>
        <ZBaseLooksItem
          :src="item.preview"
          :title="item.title"
          :path="item.path"
          :author="item.author"
          :datum="item.datum"
          @lookActivated="onLookActivated"
          @datumActivated="onDatumActivated"
        />
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
import ZBaseLooksItem from './ZBaseLooksItem.vue'

export default {
  name: 'ZBaseLooks',
  components: {
    ZBaseLooksItem
  },
  props: {
    items: {
      type: Array
    }
  },
  methods: {
    onLookActivated (item) {
      this.$emit('lookInput', item)

      this._routeTo(item.path)
    },
    onDatumActivated (item) {
      this.$emit('datumInput', item)

      this._routeTo('/datums/' + item.name)
    },
    _routeTo (path) {
      this.$router && this.$router.push(path)
    }
  },
  data () {
    return {
      size: 'lg'
    }
  }
}
</script>
