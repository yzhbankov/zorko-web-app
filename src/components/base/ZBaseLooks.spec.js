import {mount, shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'
import ZBaseLooks from './ZBaseLooks.vue'
import ZBaseLooksItem from './ZBaseLooksItem.vue'
import mockLooks from '../../../static/__mocks__/looks.json'

describe('ZBaseLooks.vue', () => {
  let propsData

  it('shows proper amount of looks', () => {
    propsData = {
      items: mockLooks
    }
    const wrapper = shallow(ZBaseLooks, {
      propsData
    })
    const homeLooksPreviewWrappers = wrapper.findAll(ZBaseLooksItem)

    expect(homeLooksPreviewWrappers.length).toEqual(8)
  })

  it('emits `openLook`', () => {
    propsData = {
      items: mockLooks
    }
    const wrapper = shallow(ZBaseLooks, {
      propsData
    })
    const {vm} = wrapper.find(ZBaseLooksItem)

    vm.$emit('lookActivated', 'bazz')

    expect(wrapper.emitted('lookInput')[0][0]).toEqual('bazz')
  })

  it('pass title to subheader', () => {
    propsData = {
      title: 'Test'
    }
    const wrapper = mount(ZBaseLooks, {
      propsData
    })
    const title = wrapper.find('.js-ZBaseLooks-subheader').text()

    expect(title).toEqual(propsData.title)
  })

  it('changes route because look`s item `lookActivated` event', () => {
    propsData = {
      items: mockLooks
    }
    const lookInfo = {path: 'bazzz'}

    const $router = {
      push: jest.fn()
    }
    const wrapper = shallow(ZBaseLooks, {
      propsData,
      mocks: {
        $router
      }
    })
    const {vm} = wrapper.find(ZBaseLooksItem)

    vm.$emit('lookActivated', lookInfo)

    expect($router.push).toHaveBeenCalledWith(lookInfo.path)
  })

  it('emits `openDatum`', () => {
    propsData = {
      items: mockLooks
    }
    const wrapper = shallow(ZBaseLooks, {
      propsData
    })
    const {vm} = wrapper.find(ZBaseLooksItem)

    vm.$emit('datumActivated', 'baz')

    expect(wrapper.emitted('datumInput')[0][0]).toEqual('baz')
  })

  it('changes route because look`s item `datumActivated` event', () => {
    propsData = {
      items: mockLooks
    }
    const $router = {
      push: jest.fn()
    }
    const datumInfo = {
      name: 'baz'
    }
    const wrapper = shallow(ZBaseLooks, {
      propsData,
      mocks: {
        $router
      }
    })
    const {vm} = wrapper.find(ZBaseLooksItem)

    vm.$emit('datumActivated', datumInfo)

    expect($router.push).toHaveBeenCalledWith('/datums/baz')
  })

  it('has same HTML for looks', () => {
    const renderer = createRenderer()
    const homeWrapper = shallow(ZBaseLooks, {
      propsData
    })

    renderer.renderToString(homeWrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
