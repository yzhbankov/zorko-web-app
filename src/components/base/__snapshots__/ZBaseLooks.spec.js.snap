// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZBaseLooks.vue has same HTML for looks 1`] = `
<v-container fluid="" grid-list-lg="true">
    <v-layout row="" wrap="">
        <v-flex xs4="">
            <!---->
        </v-flex>
        <v-flex xs4="">
            <!---->
        </v-flex>
        <v-flex xs4="">
            <!---->
        </v-flex>
        <v-flex xs4="">
            <!---->
        </v-flex>
        <v-flex xs4="">
            <!---->
        </v-flex>
        <v-flex xs4="">
            <!---->
        </v-flex>
        <v-flex xs4="">
            <!---->
        </v-flex>
        <v-flex xs4="">
            <!---->
        </v-flex>
    </v-layout>
</v-container>
`;
